ci/docker/base.dockerfile

```diff
@@ -0,0 +1,77 @@
+# Base dockerfile to build images used in Darma testing.
+
+ARG ARCH=amd64
+ARG BASE=ubuntu:22.04
+ARG SETUP_ID=${SETUP_ID:-"ubuntu-develop"}
+
+# Compiler
+ARG CC=${CC:-""}
+ARG CXX=${CXX:-""}
+ARG FC=${FC:-""}
+ARG GCOV=${GCOV:-""}
+
+# MPI
+ARG MPICH_CC=${MPICH_CC:-""}
+ARG MPICH_CXX=${MPICH_CXX:-""}
+
+# PATH
+ARG PATH_PREFIX=${PATH_PREFIX:-""}
+
+# Specific (Intel One API)
+ARG CMPLR_ROOT \
+    INTEL_LICENSE_FILE \
+    ONEAPI_ROOT \
+    TBBROOT \
+    \
+    CPATH=${CPATH:-""} \
+    INFOPATH=${INFOPATH:-""} \
+    LD_LIBRARY_PATH=${LD_LIBRARY_PATH:-""} \
+    LIBRARY_PATH=${LIBRARY_PATH:-""}
+
+FROM --platform=${ARCH} ${BASE} AS base
+
+ARG SETUP_ID CC CXX FC GCOV MPICH_CC MPICH_CXX PATH_PREFIX
+# Specific (Intel One API)
+ARG CMPLR_ROOT INTEL_LICENSE_FILE ONEAPI_ROOT TBBROOT CPATH INFOPATH LIBRARY_PATH LD_LIBRARY_PATH
+
+# Setup environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+
+ENV WF_DOCKER=1 \
+    WF_SETUP_ID=${SETUP_ID}
+
+ENV CC=$CC \
+    CXX=$CXX \
+    FC=$FC \
+    GCOV=$GCOV \
+    \
+    MPICH_CC=$MPICH_CC \
+    MPICH_CXX=$MPICH_CXX \
+    \
+    PATH="${PATH_PREFIX}${PATH}"
+
+ENV CCACHE_COMPILERCHECK=content \
+    CCACHE_COMPRESS=1  \
+    CCACHE_COMPRESSLEVE=5  \
+    CCACHE_MAXSIZE=700M \
+    CCACHE_DIR=/build/ccache \
```

Comment:
This could be placed in `ccache.conf` ([manual](https://ccache.dev/manual/latest.html#_configuration)), maybe `CCACHE_DIR` has to stay here.

Reply:
Currently it has been done as in vt by using environment variables which takes precedence over the ccache.conf.
We might add ccache.conf usage in the future but using environment variables seems is also ok and it might be more readable for projects that want to customize these values as you can see in the test-ci-project.

This could be discussed and treated in a separate issue in the future.

Reply:
Alright, let's stick with what we know works :ok_hand: